src/widgets/modals/CashTransactionModal.js

```diff
@@ -160,80 +161,79 @@ export const CashTransactionModal = ({ onConfirm }) => {
+          <View style={localStyles.textContainerStyle}>
+            <Text style={localStyles.textStyle}>{nameText}</Text>
+          </View>
+          <View style={localStyles.iconContainerStyle}>
+            <ChevronDownIcon />
+          </View>
+        </TouchableOpacity>
+        <TouchableOpacity style={localStyles.containerStyle} onPress={onPressAmount}>
+          <View style={localStyles.textContainerStyle}>
+            <Text style={localStyles.textStyle}>{amountText}</Text>
+          </View>
+          <View style={localStyles.iconContainerStyle}>
+            <PencilIcon />
+          </View>
+        </TouchableOpacity>
+        <PressReason />
+        <TouchableOpacity style={localStyles.containerStyle} onPress={onPressDescription}>
+          <View style={localStyles.textContainerStyle}>
+            <Text style={localStyles.textStyle}>{descriptionText}</Text>
+          </View>
+          <View style={localStyles.iconContainerStyle}>
+            <PencilIcon />
+          </View>
+        </TouchableOpacity>
+        <BottomCurrencyEditor
+          isOpen={isAmountModalOpen}
+          buttonText="Confirm"
+          value={amountBuffer}
+          placeholder={placeholderTextAmount}
+          onChangeText={onChangeAmount}
+          onConfirm={onSubmitAmount}
+        />
+        <BottomModalContainer
+          isOpen={isReasonModalOpen}
+          modalStyle={localStyles.bottomModalContainerStyle}
+        >
+          <GenericChoiceList
+            data={reasons}
+            keyToDisplay={CASH_TRANSACTION_KEYS.REASON}
+            onPress={onSubmitReason}
+            highlightValue={reason?.title}
+          />
+        </BottomModalContainer>
+        <BottomTextEditor
+          isOpen={isDescriptionModalOpen}
+          buttonText="Confirm"
+          value={descriptionBuffer}
+          placeholder={placeholderTextDescription}
+          onChangeText={onChangeText}
+          onConfirm={onSubmitDescription}
+        />
+        <PageButton
+          text="OK"
+          onPress={onCreate}
+          isDisabled={!isValidTransaction}
+          disabledColor={WARM_GREY}
+          style={localStyles.okButton}
+          textStyle={localStyles.pageButtonTextStyle}
         />
-      </BottomModalContainer>
-      <BottomCurrencyEditor
-        isOpen={isAmountModalOpen}
-        buttonText="Confirm"
-        value={amountBuffer}
-        placeholder={placeholderTextAmount}
-        onChangeText={onChangeAmount}
-        onConfirm={onSubmitAmount}
-      />
-      <BottomModalContainer
-        isOpen={isReasonModalOpen}
-        modalStyle={localStyles.bottomModalContainerStyle}
-      >
-        <GenericChoiceList
-          data={reasons}
-          keyToDisplay={CASH_TRANSACTION_KEYS.REASON}
-          onPress={onSubmitReason}
-          highlightValue={reason?.title}
+      </View>
+      <ModalContainer title={placeholderTextName} isVisible={isNameModalOpen}>
```

Comment:
need to provide an `onClose` prop. I think `fullScreen` would be a bit better too

Reply:
acaf42cab8a8c69fe48c990e935dbcb76a3f8529. Have omitted `fullScreen` for now, just due to the fact that the parent modal isn't fullscreen, which makes it a bit jarring. Open to changing this though.